Units/parser-cxx.r/variables-in-control-statements.d/input.cpp

```diff
@@ -0,0 +1,91 @@
+
+#include <string>
+#include <memory>
+
+// Helpers (should be hidden in ctags output)
+
+unsigned short int getUShort()
+{
+	return 10;
+}
+
+int * getPointer()
+{
+	return 0;
+}
+
+int n01;
+unsigned short n02;
+int * n03;
+
+int n04()
+{
+	return 0;
+}
+
+// The real test
+
+void test()
+{
+	// Valid variable declarations inside a for loop control structure
+	for(int f01=0;f01<10;f01++)
```

Comment:
Interesting question.
Internally the parser will tag i1 twice in the same function.
For some reason, then, a single tag will be emitted.
There must be some kind of duplicate detection mechanism somewhere in the tag file generation code.

The same happens if one declares the same variable twice in a single function.

A new feature could be to add the detection of anonymous code blocks. The local variables would be then attached to their proper scopes.